# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/GetAddrInfo.h
# Commit: 6f4ca967aa44
# Full Hash: 6f4ca967aa446cf24f4224576f51803c982f3d7b
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1901990
# File Overlap Count: 2
# Description:
#   Bug 1901990 - Introduce a pref for setting a mock HTTPS RR, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213559
# ==============================================================================

diff -r 8a2bdb44382a -r 6f4ca967aa44 netwerk/dns/GetAddrInfo.h
--- a/netwerk/dns/GetAddrInfo.h	Fri Jun 14 13:27:36 2024 +0000
+++ b/netwerk/dns/GetAddrInfo.h	Fri Jun 14 13:42:37 2024 +0000
@@ -85,6 +85,12 @@
 nsresult ParseHTTPSRecord(nsCString& aHost, DNSPacket& aDNSPacket,
                           TypeRecordResultType& aResult, uint32_t& aTTL);
 
+// Use the provided aHost to create a mock HTTPS record.
+nsresult CreateAndResolveMockHTTPSRecord(const nsACString& aHost,
+                                         uint16_t aFlags,
+                                         TypeRecordResultType& aResult,
+                                         uint32_t& aTTL);
+
 class NativeDNSResolverOverride : public nsINativeDNSResolverOverride {
   NS_DECL_THREADSAFE_ISUPPORTS
   NS_DECL_NSINATIVEDNSRESOLVEROVERRIDE